File: Week3/Assign3.1P/CounterTaskcontinued/CounterTask/Clock.cs
using System;

namespace CounterTask
{
    public class Clock
    {
        private Counter _hour;
        private Counter _minute;
        private Counter _second;

        public Clock()
        {
            _hour = new Counter("Hour");
            _minute = new Counter("Minute");
            _second = new Counter("Second");
        }

        public void Tick()
        {
            _second.Increment();
            if (_second.Ticks >= 60)
            {
                _second.Reset();
                _minute.Increment();
                if (_minute.Ticks >= 60)
                {
                    _minute.Reset();
                    _hour.Increment();
                    if (_hour.Ticks >= 24)
                    {
                        _hour.Reset();
                    }
                }
            }
        }

        public void Reset()
        {
            _hour.Reset();
            _minute.Reset();
            _second.Reset();
        }

        public string GetTime()
        {
            return $"{_hour.Ticks:D2}:{_minute.Ticks:D2}:{_second.Ticks:D2}";
        }
    }
}

File: Week11/11.1/Clock.js
class Clock {
    constructor() {
        this._hour = new Counter("Hour");
        this._minute = new Counter("Minute");
        this._second = new Counter("Second");
    }

    tick() {
        this._second.increment();
        if (this._second.ticks >= 60) {
            this._second.reset();
            this._minute.increment();
            if (this._minute.ticks >= 60) {
                this._minute.reset();
                this._hour.increment();
                if (this._hour.ticks >= 24) {
                    this._hour.reset();
                }
            }
        }
    }

    reset() {
        this._hour.reset();
        this._minute.reset();
        this._second.reset();
    }

    getTime() {
        return `${this._hour.ticks.toString().padStart(2, '0')}:${this._minute.ticks.toString().padStart(2, '0')}:${this._second.ticks.toString().padStart(2, '0')}`;
    }
}
